optimade/server/models/jsonapi.py

```diff
@@ -101,6 +101,21 @@ class Error(BaseModel):
     )
 
 
+class Warning(Error):
```

Comment:
I would put this in the `modified_jsonapi.py` file, since it is not inherently JSON API v1.0.
Also, it may be good not to overwrite the built-in `Warning` class - simply naming this class `Warnings` instead?

Furthermore, I think this can be added at the end of the class:

```python
    @validator("status")
    def status_must_not_be_specified(cls, value):
        raise AssertionError("status MUST NOT be specified for warnings")
```

Remember `from pydantic import validator` :)

---

Also, I guess it should inherit the OPTiMaDe specific `Error`?

Reply:
Done, by you!